src/status_im2/contexts/chat/messages/content/text/view.cljs

```diff
@@ -1,11 +1,105 @@
    [react-native.core :as rn]
-   [status-im.ui2.screens.chat.messages.message :as old-message]
-   [status-im2.contexts.chat.messages.link-preview.view :as link-preview]))
+   [status-im2.contexts.chat.messages.content.text.style :as style]
+   [status-im2.contexts.chat.messages.link-preview.view :as link-preview]
+   [utils.re-frame :as rf]))
+
+(defn mention-element
+  [from]
+  (rf/sub [:contacts/contact-name-by-identity from]))
+
+(defn render-inline
+  [acc {:keys [type literal destination]}]
+  (case type
+    ""
+    (conj acc literal)
+
+    "code"
+    (conj acc [rn/view {:style (merge style/block style/code)} [quo/text {:weight :code} literal]])
+
+    "emph"
+    (conj acc [quo/text {:style {:font-style :italic}} literal])
+
+    "strong"
+    (conj acc [quo/text {:weight :bold} literal])
+
+    "strong-emph"
+    (conj acc
+          [quo/text
+           {:weight :bold
+            :style  {:font-style :italic}} literal])
+
+    "del"
+    (conj acc [quo/text {:style {:text-decoration-line :line-through}} literal])
+
+    "link"
+    (conj acc
+          [quo/text
+           {:style    {:color                (colors/theme-colors colors/primary-50 colors/primary-60)
+                       :text-decoration-line :underline}
+            :on-press #(rf/dispatch [:browser.ui/message-link-pressed destination])}
+           destination])
+
+    "mention"
+    (conj
+     acc
+     [rn/touchable-opacity
+      {:active-opacity 1
+       :on-press       #(rf/dispatch [:chat.ui/show-profile literal])
+       :style          (merge style/block {:background-color colors/primary-50-opa-10})}
+      [quo/text
+       {:weight :medium
+        :style  {:color (colors/theme-colors colors/primary-50 colors/primary-60)}}
+       [mention-element literal]]])
+
+    "status-tag"
+    (conj acc
+          [quo/text
+           {:style    {:color                (colors/theme-colors colors/primary-50 colors/primary-60)
+                       :text-decoration-line :underline}
+            :on-press #(rf/dispatch [:chat.ui/start-public-chat literal])}
+           "#"
+           literal])
+
+    (conj acc literal)))
+
+
+(defn render-block
+  [acc {:keys [type ^js literal children]}]
+  (case type
+
+    "paragraph"
+    (conj acc
+          (reduce
+           (fn [acc e] (render-inline acc e))
+           [quo/text]
+           children))
+
+    "blockquote"
+    (conj acc
+          [rn/view {:style style/quote}
+           [quo/text literal]])
+
+    "codeblock"
+    (conj acc
+          [rn/view {:style (merge style/block style/code)}
+           [quo/text (.substring literal 0 (dec (.-length literal)))]])
+
+    acc))
+
+(defn render-parsed-text
+  [{:keys [content]}]
+  (reduce (fn [acc e]
+            (render-block acc e))
+          [:<>]
+          (:parsed-text content)))
 
 (defn text-content
   [message-data context]
   [rn/view
-   [old-message/render-parsed-text message-data]
+   (render-parsed-text message-data)
```

Comment:
`(` most common mistake in reagent, probably we need to add this to guidelines, cc @ilmotta

Reply:
it should be `[render-parsed-text message-data]`

Reply:
This problem you're describing is extremely important I agree, and it's well documented by Reagent and everybody should know. It even has a separate file in their docs https://github.com/reagent-project/reagent/blob/master/doc/UsingSquareBracketsInsteadOfParens.md

I believe this mistake is getting into the territory of "things everybody should know in `status-mobile`", so it's not really a guideline to me. Devs need to understand why parentheses is not okay, otherwise they'll make the same mistakes again.

Going on a tangent, have you seen https://github.com/roman01la/uix? It could be an interesting improvement over Reagent. It interprets hiccup much faster than Reagent, and it's still compatible with Reagent. Could be interesting to compare two screens (on written in Reagent and another one using uix) in mobile to see if there's any meaningful performance difference.

Reply:
But after re-reading that official doc I see there's a strong recommendation to use `[]`, but the author gives a bit of room for interpretation. I retract what I said, I'll add to the guidelines as you suggested @flexsurfer

Reply:
Just created new PR with this new recommendation https://github.com/status-im/status-mobile/pull/15008